samanthas_telegram_bot/api_queries/check.py

```diff
@@ -0,0 +1,47 @@
+"""Functions for checking existence of entities with the backend."""
+import logging
+
+import httpx
+
+from samanthas_telegram_bot.data_structures.constants import API_URL_PREFIX
+
+logger = logging.getLogger(__name__)
+# TODO check for something different in case host is unavailable? Add decorators to all functions?
```

Comment:
I advise to do general class for API calls, and put all excepting and logging to some general function like `_make_request(method, params, data, ...)`

Reply:
Done, but I still prefer to instantiate a logger here and pass to the auxiliary function so it logs the name of the module that the auxiliary function was called from - I think that will be the effect, right? :)

Reply:
Are you sure about it? Logger should put to log info about function where `logger.XXX` was called, not from where logger was instantiated. So it doesn't matter, will you pass logger somewhere or just instantiate new one, logging text will be same, and because of it it's easier to just create a new one logger for each file.

Maybe I don't understand something...

Reply:
Here's the message of my logger when I pass the logger to `log_and_notify` in `auxil` module from `PersonCreator`:
```
2023-06-01 11:21:49,978 - samanthas_telegram_bot.api_queries.person_creator - INFO - Chat ########: Created young teacher
```
If instead I drop `logger` from the args and use `logger` that I get from `logging.getLogger` inside auxil.py, this line will say `auxil.log_and_notify` instead of `api.queries.person_creator`. So, surprisingly, I seem to be correct on this one :lol: